introducing-pgzx.mdx

```diff
@@ -0,0 +1,208 @@
+---
+title: 'Introducing pgzx: create PostgreSQL extensions using Zig'
+description: 'We are excited to introduce pgzx, a framework for creating Postgres extensions using the Zig programming language.'
+image:
+  src: https://raw.githubusercontent.com/xataio/mdx-blog/main/images/introducing-pgroll.jpg
+  alt: pgzx
+author: Tudor Golubenco, Steffen Siering
+authorEmail: tudor@xata.io
+date: 3-21-2024
+tags: ['open-source', 'postgres', 'fpPgzx']
+published: true
+slug: introducing-pgzx
+ogImage: https://raw.githubusercontent.com/xataio/mdx-blog/main/images/introducing-pgroll.jpg
+---
+
+As part of the [Xata launch week](https://xata.io/launch-week-unleash-the-elephant), we are introducing [pgzx](https://github.com/xataio/pgzx), an open-source framework for developing PostgreSQL extensions written in Zig. It provides a set of utilities (e.g. error handling, memory allocators, wrappers) as well as a build and development environment that simplify integrating with the Postgres code base.
+
+## What is Zig
+
+[Zig](https://ziglang.org/) is described on its website as a general-purpose programming language and toolchain for maintaining robust, optimal and reusable software. Zig is a new language (pre 1.0), but it has been [gaining popularity](https://notes.eatonphil.com/2024-03-15-zig-rust-and-other-languages.html) in the systems programming community. It is fair to think of it as a "modern C", providing safe(r) memory management, compilation time code execution (comptime), and a rich standard library. We'll show some of these features in the context of Postgres extensions in the next sections.
+
+A major reason to use Zig for Postgres development is it's ability to interoperate with C code. Zig supports the C ABI, can work with C pointers and data types, including NULL terminated strings, and can even translate C header files into Zig code. While the automatic translating of C macros to Zig is not perfect yet, it can still help a great deal. This makes Zig a great choice for working with very large C code bases, like Postgres happens to be.
+
+## A walkthrough of pgzx
+
+Because Zig can call any C function and even C macros, you can write a Postgres extensions in Zig without any additional tooling by following the same steps you would for a C extension. However, in practice, a framework like `pgzx` simplifies the process significantly by providing a development environment, a set of utilities and wrappers for Postgres APIs, common error handling, and more.
```

Comment:
with a `*` :)

`*` call any C macro it can translate into a proper inline function.

Reply:
Ah yeah, thanks, I should clarify that.